packages/wrangler/e2e/assets-multiworker.test.ts

```diff
@@ -0,0 +1,698 @@
+
+async function fetchJson<T>(url: string, info?: RequestInit): Promise<T> {
+	return vi.waitFor(
+		async () => {
+			const text: string = await fetch(url, {
+				headers: { "MF-Disable-Pretty-Error": "true" },
+				...info,
+			}).then((r) => r.text());
+			try {
+				return JSON.parse(text) as T;
+			} catch (cause) {
+				const err = new Error(`Failed to parse JSON from:\n${text}`);
+				err.cause = cause;
+				throw err;
+			}
+		},
+		{ timeout: 5_000, interval: 250 }
+	);
+}
+
+async function startWorkersDevRegistry(
+	wranglerDev: string,
+	helper: WranglerE2ETestHelper,
+	assetWorker: string,
+	regularWorker: string,
+	regularWorkerFirst = true
+) {
+	const workerA = helper.runLongLived(wranglerDev, {
+		cwd: regularWorkerFirst ? assetWorker : regularWorker,
+	});
+	await workerA.waitForReady(5_000);
+
+	const workerB = helper.runLongLived(wranglerDev, {
+		cwd: regularWorkerFirst ? regularWorker : assetWorker,
+	});
+	const { url } = await workerB.waitForReady(5_000);
+
+	return url;
+}
+
+async function startWorkersMultiworker(
+	wranglerDev: string,
+	helper: WranglerE2ETestHelper,
+	assetWorker: string,
+	regularWorker: string,
+	regularWorkerFirst = true
+) {
+	const worker = helper.runLongLived(
+		`${wranglerDev} -c wrangler.toml -c ${regularWorkerFirst ? assetWorker : regularWorker}/wrangler.toml`,
+		{ cwd: regularWorkerFirst ? regularWorker : assetWorker }
+	);
+	const { url } = await worker.waitForReady(5_000);
+	return url;
+}
+
+const failsIf = (condition: boolean) => {
+	return condition ? it.fails : it;
+};
+
+type MultiworkerStyle = "dev registry" | "in process";
+
+describe.each([
+	{
+		style: "dev registry",
+		start: startWorkersDevRegistry,
+		wranglerDev: "wrangler dev",
+	},
+	{
+		style: "in process",
+		start: startWorkersMultiworker,
+		wranglerDev: "wrangler dev",
+	},
+	{
+		style: "dev registry",
+		start: startWorkersDevRegistry,
+		wranglerDev: "wrangler dev --x-assets-rpc",
+	},
+	{
+		style: "in process",
+		start: startWorkersMultiworker,
+		wranglerDev: "wrangler dev --x-assets-rpc",
+	},
+] as {
+	style: MultiworkerStyle;
+	start: typeof startWorkersDevRegistry | typeof startWorkersMultiworker;
+	wranglerDev: string;
+}[])(
+	"workers with assets ($style, $wranglerDev) ",
+	async ({ start, style, wranglerDev }) => {
+		let assetWorkerName: string;
+		let regularWorkerName: string;
+		let assetWorker: string;
+		let regularWorker: string;
+		let helper: WranglerE2ETestHelper;
+
+		beforeEach(async () => {
+			helper = new WranglerE2ETestHelper();
+
+			assetWorker = await makeRoot();
+			assetWorkerName = generateResourceName("worker-aw");
+			regularWorkerName = generateResourceName("worker");
+			await baseSeed(assetWorker, {
+				"wrangler.toml": dedent`
+						name = "${assetWorkerName}"
+						main = "src/index.ts"
+						compatibility_date = "2024-11-01"
+
+						[assets]
+						directory = "./public/"
+						binding = "ASSETS"
+
+						[[services]]
+						binding = "DEE"
```

Comment:
maybe :point-down: since `DEE` is not very descriptive?
```suggestion
						binding = "DEFAULT_EXPORT"
```